src/Chainweb/Miner/Coordinator.hs

```diff
@@ -157,17 +160,23 @@ newWork logFun choice miner pact c = do
 
             pure $ T3 (PrevTime $ _blockCreationTime p) header payload
 
-chainChoice :: Cut -> ChainChoice -> IO ChainId
-chainChoice c choice = case choice of
-    Anything -> randomChainId c
+chainChoice :: TVar [ChainId] -> ChainChoice -> IO ChainId
+chainChoice tchains choice = case choice of
+    Anything -> nextChainId tchains
     Suggestion cid -> pure cid
     TriedLast cid -> loop cid
   where
     loop :: ChainId -> IO ChainId
     loop cid = do
-      new <- randomChainId c
+      new <- nextChainId tchains
```

Comment:
I am a bit uncomfortable with this change. This means all chains are visited in the same order by all nodes. Sure, different nodes should mine at different chains, but, if for instance many chains become blocked, all nodes would synchronize and start in the same order.

Usually, uniform random choice is a powerful algorithmic tool that enforces many desired properties at very low cost (assuming availability of a reliable random source). In particular it is efficient in mitigating many attacks. 

At the very least I propose that each node picks a random permutation of the chains at startup.

Reply:
Since this is at the core of Chainweb mining strategy: @buckie what are your thoughts on this?

Reply:
In practice this list will progress _very_ quickly, and also at different rates on each node that is doing coordination. The nodes won't in any way be synchronized.

Reply:
Yes we have to shuffle the list on startup. :+1:

Reply:
Actually, I don't think that's enough. In the beginning there will be relatively few coordinator nodes serving many miners.

Reply:
Also this approach only works for < 10 miners. Beyond that it's not better than random sampling, which is safer.